text/0000-cargo-workspace-deduplicate.md

```diff
@@ -0,0 +1,630 @@
+dependencies. This part comes later below.
+
+The `[workspace]` section will also allow the definition of a number of keys
+also defined in `[package]` today, namely:
+
+```toml
+[workspace]
+authors = ["Nice Folks"]
+description = "..."
+documentation = "https://example.github.io/example"
+readme = "README.md"
+homepage = "https://example.com"
+repository = "https://github.com/example/example"
+license = "MIT"
+license-file = "./LICENSE"
+keywords = ["cli"]
+categories = ["development-tools"]
+publish = false
+```
+
+Each of these keys have no meaning in a `[workspace]` table yet, but will have
+meaning when they're assigned to crates internally. That part comes later though
+in this design! Note that the format and accepted values for these keys are the
+same as the `[package]` section of `Cargo.toml`.
+
+Finally, the `[workspace.badges]` table can also be specified and mirrors the
+`[badges]` table per-package.
+
+For now keys such as `edition` and `metadata` are explicitly left out, but they
+can always be added in the future if necessary.
+
+## Updates to a package `Cargo.toml`
+
+The interpretation of a `Cargo.toml` manifest within Cargo will now require a
+`Workspace` object to be created. This `Workspace` will be used to elaborate and
+expand each member's `Cargo.toml` directive. Additionally `Cargo.toml` will
+syntactically accept some more forms.
+
+### Placeholder Versions
+
+Previously the version of a package was specified as:
+
+```toml
+[package]
+version = "1.2.3"
+```
+
+Cargo will now accept a table definition of `package.version` which defines the
+`package.version.workspace` key as well. The `package.version.workspace` key
+must be a string, and it must name a package present in the workspace
+elsewhere. For example:
+
+```toml
+[package]
+name = "foo"
+version = { workspace = "other" }
+```
+
+This directive indicates that the version of `foo` is the same as the version of
+the workspace member `other`. Cyclic references between packages for the version
+are disallowed and will result in an error from Cargo. A package also cannot
+reference itself for its own version.
+
+### Metadata defaults to workspace metadata
+
+A number of metadata keys in `[package]` are not required to be defined, and if
+they are not defined they will default to the value in `[workspace]` if it is
+present there. If no `[workspace]` is present or the `[workspace]` section
+doesn't define these keys then their values will remain undefined as they are
+today.
+
+The following keys in `[package]` are now inferred to be the value in
+`[workspace]` if the value in `[package]` isn't present and the value in
+`[workspace]` is.
+
+```toml
+[package]
+authors = ["Nice Folks"]
+license = "MIT"
+license-file = "./LICENSE"
+repository = "https://github.com/example/example"
+documentation = "https://example.github.io/example"
+categories = ["development-tools"]
+keywords = ["cli"]
+publish = false
+```
```

Comment:
I think this list needs to be synced with the one above (add homepage and readme and maybe description).